ops/testing.py

```diff
@@ -1530,6 +1525,42 @@ def add_model_secret(self, owner: AppUnitOrName, content: Dict[str, str]) -> str
         model.Secret._validate_content(content)
         return self._backend._secret_add(content, owner_name)
 
+    def add_model_secret(self, owner: AppUnitOrName, content: Dict[str, str]) -> str:
+        """An alias for :meth:`add_charm_secret` for backward compatibility."""
+        warnings.warn("add_model_secret is going to be deprecated, use add_charm_secret instead.",
+                      category=PendingDeprecationWarning)
+        return self.add_charm_secret(owner, content)
+
+    def add_user_secret(self, content: Dict[str, str]) -> str:
+        """Add a secret owned by the user, simulating the `juju add-secret` command.
+
+        Although it's named a user-owned secret in juju, technically, the owner is the
+        Model, so the secret's owner is set to :attr:`Model.uuid <ops.Model.uuid>`.
+
+        Args:
+            content: A key-value mapping containing the payload of the secret,
+                for example :code:`{"password": "foo123"}`.
+
+        Return:
+            The ID of the newly-secret added.
+
+        Example usage (the parameter ``harness`` in the test function is
+        a pytest fixture that does setup/teardown, see :class:`Harness`)::
+
+            # test_charm.py
+            def test_start(harness):
+                secret_content = {'password': 'foo'}
+                secret_id = harness.add_user_secret(secret_content)
+                # grant the user secret to the app
+                harness.grant_secret(secret_id, 'webapp')
+                secret = harness.model.get_secret(id=secret_id)
+                assert secret.id == secret_id
+                assert secret.get_content() == secret_content
```

Comment:
This feels a bit artificial - in reality you'd have a charm that did something with the secret, and since the charm can't make any changes to the secret you wouldn't ever be asserting on it afterwards.

However, I'm not sure how detailed we want these little examples in the API reference docs to be - a more realistic example is perhaps better suited to a how-to guide, so maybe leave this as-is?

Reply:
Yeah, it feels totally made-up and has no real-world guidance value at all. Let me whip up a useful usecase.